src/react/utils.js

```diff
@@ -58,6 +60,92 @@ export function valueIsClassComponent(realm: Realm, value: Value) {
   return false;
 }
 
+function canHoistObject(realm: Realm, object: ObjectValue, hoistedScope?: Generator): boolean {
```

Comment:
Could you put a comment that describes when things should and shouldn't be hoisted.